backend/src/apiserver/server/pipeline_server.go

```diff
@@ -98,427 +108,1027 @@ type PipelineServer struct {
 	options         *PipelineServerOptions
 }
 
-func (s *PipelineServer) CreatePipelineV1(ctx context.Context, request *api.CreatePipelineRequest) (*api.Pipeline, error) {
-	if s.options.CollectMetrics {
-		createPipelineRequests.Inc()
+func NewPipelineServer(resourceManager *resource.ResourceManager, options *PipelineServerOptions) *PipelineServer {
+	return &PipelineServer{resourceManager: resourceManager, httpClient: http.DefaultClient, options: options}
+}
+
+// This method extract the common logic of naming the pipeline.
+// API caller can either explicitly name the pipeline through query string ?name=foobar.
+// or API server can use the file name by default.
+func getPipelineName(queryString string, fileName string) (string, error) {
```

Comment:
The purpose of the function and the name of the function do not match.

Reply:
Thank you for pointing this out. This function was created earlier, but renaming makes sense.